File: domasna1/infobank/src/main/java/mk/finki/ukim/infobank/Model/BankEntity.java
package mk.finki.ukim.infobank.Model;

import lombok.Data;
import org.springframework.data.annotation.Id;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

@Data
public class BankEntity {
    @Id
    private String id;
    private String type;
    private String name;
    private double lon;
    private double lat;
    private Map<String,String> details;

    public BankEntity(Map<String,String> data) {
        HashMap<String,String> detailsCopy = new HashMap<>(data);
        id = detailsCopy.remove("id");
        name = detailsCopy.remove("name");
        type = detailsCopy.remove("amenity");
        lon = Double.parseDouble(detailsCopy.get("lon"));
        lat = Double.parseDouble(detailsCopy.get("lat"));
        details = detailsCopy;
        detailsCopy.remove("lat");
        detailsCopy.remove("lon");
    }
}

File: Domasno4/infobank-frontend/src/app/models/BankEntity.ts
export class BankEntity {
  public id: string;
  public _class: string;
  public lat: number;
  public lon: number;
  public name: string;
  public type: string;
  public details:  { [key: string]: string } = {};

  constructor(id: string, _class: string, lat: number, lon: number, name: string, type: string, details: {}) {
    this.id = id;
    this._class = _class;
    this.lat = lat;
    this.lon = lon;
    this.name = name;
    this.type = type;
    this.details = details;
  }
}
